otel-agent/ecs-ec2/config.yaml

```diff
@@ -32,8 +32,8 @@ processors:
   batch:
   batch/metrics:
     timeout: 60s
-  resourcedetection/ecs:
-    detectors: [env, ecs]
+  resourcedetection:
+    detectors: [env, ec2, ecs, docker, system]
```

Comment:
Do we want to use both `docker` and `system`? As far as I understood the [docs](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/resourcedetectionprocessor/README.md#system-metadata), both give you the same info (host name and OS), but you should use Docker if running collector as Docker container.

Reply:
Not sure why but in testing, when I removed system, the **host.id** would be missing from the resource attributes. I can check the further to be sure.

Reply:
I guess then we could use `system` only? Not sure how important / usable the `host.id` information is when running in ECS. Docker does not provide this one I think.

Reply:
I believe host.id and host.name, etc are required for APM. Also note removing `docker` will also make the the `host.id` and other attributes disappear.

The otel docs on this seem misleading.

Attributes **without** `system`:
<img width="835" alt="Screenshot 2023-03-22 at 11 21 34" src="https://user-images.githubusercontent.com/4018554/226881655-6ecfd842-5c37-4dfc-89f0-95fe2ce212d3.png">

Attributes **with** `system`
<img width="790" alt="Screenshot 2023-03-22 at 11 43 42" src="https://user-images.githubusercontent.com/4018554/226881693-10b84eac-c7fa-4306-bac2-cc2f73b1c019.png">

Reply:
Thanks for the detailed explanation! Let's keep it then.